affectedmanifests.go

```diff
@@ -0,0 +1,58 @@
+package claircore
+
+import (
+	"sort"
+	"sync"
+)
+
+// AffectedManifests describes a set of manifests affected by
+// a set of Vulnerabilities.
+type AffectedManifests struct {
+	mu sync.Mutex
+	// map of vulnerabilities keyed by the vulnerability's ID
+	Vulnerabilities map[string]Vulnerability `json:"vulnerabilities"`
```

Comment:
The value should be `*Vulnerability`, otherwise any growing the map will cause a *lot* of copying.

Reply:
should I be concerned with polluting the heap vs copy speed here?

Reply:
I mean, we have to get the Vulnerability from somewhere to pass in. The entrypoint into the affected manifests has them in a slice, but then ends up copying them all around. This and the `Add` method should just take pointers, because the caller has packed them into a slice for us.

Reply:
I think I see what you're saying, either way I am making copies when Add is called, those copies may as well be pointers, and then when the map needs to grow it's less expensive.